### ISSUE TITLE
[CP] Cherry pick did not make it into the stable release

### ISSUE THREAD

**Issue by cbenhagen** [flutter/flutter#111808]
For some reason the cherry pick requested in #109233 did not yet make it into the stable channel.
Original issue: #108737

cc @zanderso 

## Steps to reproduce
1. Execute `flutter create package_test && cd package_test && flutter build macos`
2. Check *.frag file permissions with `find . -name "*.frag" -exec stat -f "%A %N" {} \;`

### Expected results ###
Files to be included in the bundle have read permissions for all users.

### Actual results ###
600 permissions on files:
```
600 ./build/macos/Build/Products/Release/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag
600 ./build/macos/Build/Products/Release/package_test.app/Contents/Frameworks/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag
```

Tested on:
`Flutter (Channel stable, 3.3.2, on macOS 12.6 21G115 darwin-arm, locale en-CH)`

**Comment by huycozy** [comment]
Hi @cbenhagen, thanks for filing the issue. This issue is reproducible on the latest stable `3.3.2` and beta `3.3.0-0.5.pre` but it doesn't happen on the latest master channel `3.4.0-19.0.pre.337`.

- stable: ❌
```
➜  test_0644_macos_111808 find . -name "*.frag" -exec stat -f "%A %N" {} \;
600 ./build/macos/Build/Products/Release/test_0644_macos_111808.app/Contents/Frameworks/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag
600 ./build/macos/Build/Products/Release/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag
```

- beta: ❌
```
➜  test_0644_macos_111808_beta find . -name "*.frag" -exec stat -f "%A %N" {} \;
600 ./build/macos/Build/Products/Release/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag
600 ./build/macos/Build/Products/Release/test_0644_macos_111808_beta.app/Contents/Frameworks/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag
```

- master: ✅
```
➜  test_0644_macos_111808_master find . -name "*.frag" -exec stat -f "%A %N" {} \;
644 ./build/macos/Build/Products/Release/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag
644 ./build/macos/Build/Products/Release/test_0644_macos_111808_master.app/Contents/Frameworks/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag
```

✅: No Issue ❌: Issue reproduced

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 3.3.2, on macOS 12.5.1 21G83 darwin-x64, locale en-VN)
    • Flutter version 3.3.2 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e3c29ec00c (12 hours ago), 2022-09-14 08:46:55 -0500
    • Engine revision a4ff2c53d8
    • Dart version 2.18.1
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13E113
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.71.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.5.1 21G83 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```bash
[✓] Flutter (Channel master, 3.4.0-19.0.pre.337, on macOS 12.5.1 21G83 darwin-x64, locale en-VN)
    • Flutter version 3.4.0-19.0.pre.337 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 085a187e04 (75 minutes ago), 2022-09-18 21:11:34 -0400
    • Engine revision b22a795011
    • Dart version 2.19.0 (build 2.19.0-214.0.dev)
    • DevTools version 2.17.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13E113
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.71.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 12 (API 31)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 12.5.1 21G83 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

```bash
[✓] Flutter (Channel beta, 3.3.0-0.5.pre, on macOS 12.5 21G72 darwin-x64, locale en-EE)
    • Flutter version 3.3.0-0.5.pre on channel beta at /Users/huynq/Documents/GitHub/flutter_beta
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 096162697a (8 days ago), 2022-08-22 15:34:14 -0700
    • Engine revision ad3d868e0d
    • Dart version 2.18.0 (build 2.18.0-271.7.beta)
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13E113
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2020.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 12 (API 31)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 12.5 21G72 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 104.0.5112.101

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```
</details>

Labeling this issue for further investigation. Thank you.

**Comment by zanderso** [comment]
Routing to @itsjustkevin to verify that the cherry-pick was applied, and that the release used the engine built with the applied cherry-pick.

**Comment by zanderso** [comment]
@itsjustkevin the CP pull request was here https://github.com/flutter/engine/pull/35269. Do you see anything wrong with how it was landed? Is it against the right branch? Have we been re-building the engine branch for the hotfix releases?

**Comment by itsjustkevin** [comment]
@zanderso  This was cherry-picked to the beta branch last month and promoted to stable.  Could we be missing an associated cherry-pick?

**Comment by zanderso** [comment]
I have not investigated fully, but since this is working on master, then it is likely that the fix relied on a toolchain or libstdcxx update, which would be risky to cherry-pick into stable. My understanding is that it is possible to address this issue by manually fixing the file permissions. In that case, on balance, I don't think the fix is worth the risk.

**Comment by deimantasa** [comment]
Until it lands properly to `stable`, sharing some detailed tutorial for folks, who are running some CI/CD tools and might facing this issue. 

**TL;DR** permissions will be given to the right file once build is finished automatically via xcode post build script.

1. Go to Xcode
2. Open your macos project
3. Product -> Scheme -> Edit Scheme
4. Build -> Post-actions
5. Select your `Runner`
6. Enter script 
```
# Due to macos flutter build issue, some permissions are wrong. Update manually.
# Read more here https://github.com/flutter/flutter/issues/108737 and https://github.com/flutter/flutter/issues/111808
# Delete once it lands on stable
chmod 0644 "$TARGET_BUILD_DIR/$FULL_PRODUCT_NAME/Contents/Frameworks/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag"
```
7. Hit `Close`

![temp-fix-permission-issue](https://user-images.githubusercontent.com/12739071/193439874-cfd29939-1906-435a-b622-4c27c3ed12f7.png)

Now if you build macos app, once build is finished, this script will execute and correct permissions will be given.

**Comment by krburkhart** [comment]
I have noticed this issue on a Linux build with flutter 3.3.4.  It was exposed by a backup program that requires files to be group readable.  Will the file mode changes in this fix be done on Linux builds as well?
